drivers/sensor/max6675/max6675.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2021 Teslabs Engineering S.L.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT maxim_max6675
+
+#include <drivers/sensor.h>
+#include <drivers/spi.h>
+#include <sys/byteorder.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(max6675, CONFIG_SENSOR_LOG_LEVEL);
+
+/** Thermocouple input bit (for NC detection). */
+#define THERMOCOUPLE_INPUT BIT(2)
+/** Temperature position. */
+#define TEMPERATURE_POS 3U
+/** Temperature resolution (cDeg/LSB). */
+#define TEMPERATURE_RES 25U
+
+struct max6675_config {
+	const char *spi_label;
+	uint16_t spi_addr;
+	uint32_t spi_max_freq;
+	const char *spi_cs_label;
+	gpio_pin_t spi_cs_pin;
+	gpio_dt_flags_t spi_cs_flags;
+};
+
+struct max6675_data {
+	const struct device *spi;
+	struct spi_config spi_config;
+	struct spi_cs_control cs_ctrl;
+	uint16_t sample;
+};
+
+static int max6675_sample_fetch(const struct device *dev,
+				enum sensor_channel chan)
+{
+	struct max6675_data *data = dev->data;
+
+	int ret;
+	uint8_t buf_rx[2];
+	const struct spi_buf rx_buf = {
+		.buf = buf_rx,
+		.len = ARRAY_SIZE(buf_rx)
+	};
+	const struct spi_buf_set rx_bufs = {
+		.buffers = &rx_buf,
+		.count = 1U
+	};
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	ret = spi_read(data->spi, &data->spi_config, &rx_bufs);
+	if (ret < 0) {
+		return ret;
+	}
+
+	data->sample = sys_get_be16(buf_rx);
+
+	if (data->sample & THERMOCOUPLE_INPUT) {
+		LOG_ERR("Thermocouple not connected");
+		return -ENODEV;
```

Comment:
In similar applications with thermistors I return an out-of-band value that indicates the thermistor isn't present, e.g. -300 Cel.  That wouldn't fit with the existing sensor API, but I do think it's worth distinguishing a bus error from an I/O error; maybe `-ENOENT` to indicate the thermocouple isn't present?  `-ENODEV` seems wrong since the sensor device itself is present.

I'd probably go with `LOG_INF`.  A distinct error value will tell the application what's wrong.  (Sadly the sensor API doesn't provide information on failure modes indicated by negative return values.)

Reply:
I've opted to change log to `LOG_INF` and report `-ENOENT`. I think that `get` should not be called if fetch fails, so I prefer to skip returning a special out-of-band value.